### PR TITLE
Adds distributed UNet example

### PR DESCRIPTION
**Pull request by dmsuehir**
This PR adds an examples directory, with the distributed UNet use case as our first example.  I have a README that explains the steps on running the example (which involves using KVC and MLT) as well as information on the steps that I took to go from an MLT template to running the UNet model.

## Reviewer Checklist

 - [x] Do you understand it?
 - [x] Is it correct?
 - [x] Is it safe?
 - [x] Is it legally compliant?
 - [x] Is it robust?
 - [x] Is it simple?
 - [x] Is it tested?
 - [x] Is it documented?
 - [x] Is the style consistent?

See [the reviewer guide](docs/reviews.md) for more detail on each check.

### CODE REVIEW

**Comment by karthikvadla** [comment]
One more thing, did you add K8s directory . It complied about that while deploying . Otherthan that everything looks good .

**Comment by karthikvadla** [comment]
Just saw you already added k8s directory . Looks good to me .

File: examples/distributed_unet/README.md [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+    `nodeAffinity` and `hostPath` came from the KVC `volumemanager` that
+    was setup earlier.
+ 4. Updated the `settings_dist.py` file to get the `OUTPUT_PATH`
+ environment variable for storing checkpoint files.
+ 5. Updated the main `test_dist.py` file to get the `TF_CONFIG`
+ environment variable for the cluster information (list of workers,
+ parameter server, job name, and task index).
+
+The updated app is located in the [mlt_app](mlt_app) directory.  Before
+running the app, you will need to update:
```
[the remark this answers is not in the document]
Comment (dmsuehir) [reply examples/distributed_unet/README.md]: Yes, I added the mlt.json.  thanks

**Comment by dmsuehir** [comment]
@ashahba Should I add license headers to the UNet code files and remove the `LICENSE` file?  I grabbed the UNet model from here: https://github.com/NervanaSystems/topologies/tree/master/distributed_unet
Where they don't have the headers, but they have a `LICENSE` file in that directory.

**Comment by ashahba** [comment]
@dmsuehir let me review and I get back to you on this.

File: examples/distributed_unet/README.md [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+to transfer the data to the nodes prior to running model training.
+
+## Prerequisites
+
+This example requires that you have MLT and it's depedencies installed,
+as well as the following CRDs are installed on your Kubernetes cluster:
+- [Kube-Volume-Controller](https://github.com/kubeflow/experimental-kvc/blob/master/docs/ops.md#installing-the-controller)
+- [TF-Operator](https://github.com/kubeflow/tf-operator)
+
+You will also need to register to obtain the BraTS dataset.  Information
```
Comment (nqn) [inline examples/distributed_unet/README.md]: Nit: maybe move "Information" to the next line

File: examples/distributed_unet/README.md [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+Update this file with your information (such as your `namespace`,
+`sourceURL`, etc), then use the following command to create the KVC
+Volume Manager for the dataset:
+
+```bash
+$ kubectl create -f kvc/kvc_unet_gcs.yaml
+volumemanager "unet-data" created
+```
+
+Once the data transfer has completed, the `volumemanager` has
```
Comment (nqn) [inline examples/distributed_unet/README.md]: 'has `message: success`'? What does that mean :) in the status?

File: examples/distributed_unet/README.md [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+`message: success`.  We can then get the `nodeAffinity` information,
```
Comment (nqn) [inline examples/distributed_unet/README.md]: Nit: could consider to drop 'information' so it sounds like "get the node affinity"

File: examples/distributed_unet/mlt_app/k8s-templates/tfjob.yaml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+          restartPolicy: OnFailure
+          affinity:
+            nodeAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+                nodeSelectorTerms:
+                - matchExpressions:
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    - gke-dls-us-n1-standard-4-1ba3a893-516b
```
Comment (nqn) [inline examples/distributed_unet/mlt_app/k8s-templates/tfjob.yaml]: These are internal server names - can we replace them with placeholders?

File: examples/distributed_unet/mlt_app/k8s-templates/tfjob.yaml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+          restartPolicy: Never
+          affinity:
+            nodeAffinity:
+              requiredDuringSchedulingIgnoredDuringExecution:
+                nodeSelectorTerms:
+                - matchExpressions:
+                  - key: kubernetes.io/hostname
+                    operator: In
+                    values:
+                    - gke-dls-us-n1-highmem-8-skylake-82af83b4-7gc4
```
Comment (nqn) [inline examples/distributed_unet/mlt_app/k8s-templates/tfjob.yaml]: Same here ^^

File: examples/distributed_unet/mlt_app/model.py [hunk 7]
```diff
@@ -0,0 +1,175 @@
+import os.path
```
Comment (nqn) [inline examples/distributed_unet/mlt_app/model.py]: Do we need license headers here?

File: examples/distributed_unet/mlt_app/preprocess.py [hunk 8]
```diff
@@ -0,0 +1,399 @@
+""" To conver dicom images into images needed for keras"""
```
Comment (nqn) [inline examples/distributed_unet/mlt_app/preprocess.py]: Same here - license headers?

**Comment by ashahba** [comment]
@dmsuehir If we changes any of the files in:
https://github.com/NervanaSystems/topologies open to public?

Then for the files we changed, we need to have both license headers (in case the licenses are different). Ours on top followed by the original header.

And technically we don't need to keep that `LICENSE` since we already have a exact copy of it shipped with our code so end users already have one.

But based on what's under `Citations` for that repo, if we used `BraTS` data, we need to cite it.

Also as @nqn mentioned above we need to make server names anonymized like for example:
```gke-mycluster-mynode-82af83b4-7gc4```

**Comment by dmsuehir** [comment]
Thanks @ashahba.  I'll add the license header and will remove the `LICENSE` file.  I have the BraTS Citation already -- at the bottom of the `README.md`.

**Comment by dmsuehir** [comment]
@ashahba @nqn  I made the updates you requested.  Added license headers and modified the server names to be like `gke-cluster-n1-highmem-8-skylake-1`, `gke-cluster-n1-highmem-8-skylake-2`, etc., and made README file updates suggested by Niklas (including adding a link to the example from the main README).

**Pull request merged**